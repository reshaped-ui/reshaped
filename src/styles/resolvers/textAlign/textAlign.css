@responsive [style*="--rs-text-align-s:"] {
	@variable --rs-text-align;

	text-align: var(--rs-text-align) !important;
}
