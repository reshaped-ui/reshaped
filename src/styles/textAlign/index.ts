import { responsiveClassNames } from "utilities/helpers";
import * as T from "styles/types";
import s from "./textAlign.module.css";

const getTextAlignStyles: T.StaticStyleUtility<T.TextAlign> = (value) => {
	if (!value) return null;

	return {
		classNames: [...responsiveClassNames(s, "--text-align", value)],
	};
};

export default getTextAlignStyles;
