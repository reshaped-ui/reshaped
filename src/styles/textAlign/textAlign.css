@responsive [style*="--rs-text-align-"] {
	@variable --rs-text-align;

	text-align: var(--rs-text-align) !important;
}
