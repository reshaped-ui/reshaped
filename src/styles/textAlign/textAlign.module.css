@responsive .--text-align {
	@value start {
		text-align: start;
	}

	@value center {
		text-align: center;
	}

	@value end {
		text-align: end;
	}
}
