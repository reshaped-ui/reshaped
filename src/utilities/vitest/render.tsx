import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
