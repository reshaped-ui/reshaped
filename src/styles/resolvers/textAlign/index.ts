import { responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import "./textAlign.css";

const textAlign: T.StyleResolver<T.TextAlign> = (value) => {
	if (!value) return {};

	return {
		variables: responsiveVariables("--rs-text-align", value),
	};
};

export default textAlign;
