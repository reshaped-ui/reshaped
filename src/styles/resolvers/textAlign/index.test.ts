import { expect, test, describe } from "vitest";
import textAlign from "./index";

describe("Styles/TextAlign", () => {
	test("handles value", () => {
		expect(textAlign("start")).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(textAlign()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(textAlign({ s: "start", m: "center", l: "end" })).toMatchSnapshot();
	});
});
