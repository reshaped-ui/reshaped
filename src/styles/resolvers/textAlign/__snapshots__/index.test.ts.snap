// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/TextAlign > handles responsive value 1`] = `
{
  "variables": {
    "--rs-text-align-l": "end",
    "--rs-text-align-m": "center",
    "--rs-text-align-s": "start",
  },
}
`;

exports[`Styles/TextAlign > handles undefined value 1`] = `{}`;

exports[`Styles/TextAlign > handles value 1`] = `
{
  "variables": {
    "--rs-text-align-s": "start",
  },
}
`;
